python/cudf/cudf/_lib/csv.pyx

```diff
@@ -273,8 +273,7 @@ def read_csv(
         elif isinstance(dtype, abc.Collection):
             for index, col_dtype in enumerate(dtype):
                 if isinstance(cudf.dtype(col_dtype), cudf.CategoricalDtype):
-                    col_name = df._data.names[index]
-                    df._data[col_name] = df._data[col_name].astype(col_dtype)
+                    df.iloc[:, index] = df.iloc[:, index].astype(col_dtype)
```

Comment:
Will this change not end up being expensive since we are touching the public dataframe API + typecasting a series which will be having the returning the index too?

Reply:
Yeah that's fair. I'll revert this change